rwkv.cpp

```diff
@@ -600,6 +605,45 @@ struct rwkv_context * rwkv_init_from_file(const char * file_path, const uint32_t
         RWKV_ASSERT_NULL(RWKV_ERROR_MODEL_PARAMS, set_block_parameter(&parameters, i, "ffn.receptance.weight", &layer->ffn_receptance));
     }
 
+#ifdef GGML_USE_CUBLAS
+    {
+        const int n_gpu = std::min(n_gpu_layers, model->n_layer);
+
+        fprintf(stdout, "%s: [cuBLAS] offloading %d layers to GPU\n", __func__, n_gpu);
+
+        size_t vram_total = 0;
+
+        for (int i = 0; i < n_gpu; ++i) {
```

Comment:
(just asking) Why specific layers are moved to GPU, as compared to move each weight matrix to GPU just before multiplying it and removing it from GPU?

Reply:
> as compared to move each weight matrix to GPU just before multiplying it and removing it from GPU

sending messages/memory to gpu is slow as it is asynchronous

Reply:
Best solution should be loading the gpu layers directly to gpu even without copying them to memory once, which is optimized on llama.cpp. I was plain to implement the feature once there is any requirement